interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a JavaSript libray made up of a collection of code snippets used for building user interfaces. Code is reusable and handles user interactions. it is also scalable which allows you to think about how your app will grow.
 
-  Researched answer:
+  Researched answer: React allows for developers to create large web applications. Allows for data to be manipulated without reloading a web page when a user is interacting with the interface. The virtual DOM is listening for a diff. Reusable components such as classes.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: Anytime you are creating a class component, import which allows using contents from another file, define the class and extend component. render method, return and export which makes files contents elgible for importing.
 
-  Researched answer:
+  Researched answer: the lifecycle of components which can be manipulated during the three main phases are mounting, updating and unmounting.contructor method and render method are part of the mounting phase, updating phase include any updates made to the class componenet. when a componenet is removed from the DOM the phase is referred to to as the unmounting phase.
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: My understanding of JSX is that its a markdown language combined of HTML and JavaScript. When typing JSX language in react it is required to use curly braces in order for JavaScript language to be read properly within the HTML code.
 
-  Researched answer:
+  Researched answer: I would say its like a template language. The difference in syntax is the combination of using JavaScript code and curly braces with HTML tags. JSX can used to specify attributes can also be used inside of `if statements`, `for loops`. JSX can also be assigned to `vairables`,arguments and be part of a `return` in a function.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a JavaScript package manager that creates the file structure of a React app.Yarn also organizes and manages the node modules.
 
-  Researched answer:
+  Researched answer: Yarn manages files in node.js-open source, cross platform used on the front and back end, also uses files like yarn.lock-main source of information about a current project and package.json-holds metadate relevant to the project.
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: a function that is not named or does not have value, High order functions use anonymous functions?
 
-  Researched answer:
+  Researched answer: Anonymouse functions can be called upon by other functions like high order functions which takes a function as an argument and anonymous functions do not require a name.
```

Comment:
Excellent work in this section!  You seem to have a firm understanding of the vocabulary!